extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/DerivedMethodsAdder.java

```diff
@@ -340,7 +340,7 @@ private void generateCustomResultTypes(DotName interfaceName, DotName implName,
                     throw new IllegalArgumentException("Method " + method.name() + " of interface " + interfaceName
                             + " is not a getter method since it returns void");
                 }
-                DotName fieldTypeName = getPrimitiveTypeName(returnType.name());
+                DotName fieldTypeName = returnType.name();
```

Comment:
I don't see how this change is related to the linked issue.
Also, the `getPrimitiveTypeName` method is still used in `CustomQueryMethodsAdder`.

Reply:
It's related to the reproducer. This method should not have been called here

Reply:
So, it's expected not to be called here, but it is in CustomQueryMethodsAdder?
Not familiar with the code, so I trust on your judgement :)

Reply:
> So, it's expected not to be called here, but it is in CustomQueryMethodsAdder?

I honestly don't remember, so for the time being I'll leave it as it is :)